api/infoGatherer/specDetails.go

```diff
@@ -37,14 +37,28 @@ import (
 // SpecInfoGatherer contains the caches for specs, concepts, and steps
 type SpecInfoGatherer struct {
 	waitGroup         sync.WaitGroup
-	mutex             sync.Mutex
 	conceptDictionary *gauge.ConceptDictionary
-	specsCache        map[string]*SpecDetail
-	conceptsCache     map[string][]*gauge.Concept
-	stepsCache        map[string][]*gauge.StepValue
+	specsCache        specsCache
+	conceptsCache     conceptCache
+	stepsCache        stepsCache
 	SpecDirs          []string
 }
 
+type conceptCache struct {
+	mutex         sync.RWMutex
+	conceptsCache map[string][]*gauge.Concept
```

Comment:
perhaps this field should have a different name? this leads to access such as `conceptCache.conceptsCache` below..

Similar for `stepsCache` and `specsCache`. Perhaps the name of the entity to be used makes sense? i.e. `conceptCache.concepts`?